src/server.py

```diff
@@ -1,28 +1,61 @@
 import asyncio
+import os
 
+import boto3
 import websockets
 from aiohttp import web
+from dotenv import load_dotenv
 
-from cuda_utils import cuda_device_enable
+from app import add_model_handler, activate_model_handler, delete_model_handler, list_models_handler, model_service
 from model_operations import generate_tokens, load_model
 
-# Ensure the model is loaded on the GPU
-print(f"device set to: {cuda_device_enable()}")
+load_dotenv()
 
-# Load the model and tokenizer
-model_name = os.environ['MODEL_REPO']
-model_file = os.environ['MODEL_FILE']
-model_path = f"model/{model_file}"
-print("Loading model")
-model = load_model(model_path=model_path, n_ctx=4096)
-print("Server ready")
+
+def download_model_from_s3(s3_path, local_path):
+    """
+    Downloads the model from S3 if it's not already present locally.
+    """
+    s3 = boto3.client(
+        's3',
+        aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID"),
+        aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY"),
+        region_name="us-east-1"
+    )
```

Comment:
I don't think the `server` should know about `S3` or how the model is downloaded.